pkg/snapshot/v1alpha1/snapshot_test.go

```diff
@@ -0,0 +1,175 @@
+package snapshot
+
+import (
+	"os"
+	"testing"
+
+	"github.com/openebs/maya/pkg/apis/openebs.io/v1alpha1"
+	menv "github.com/openebs/maya/pkg/env/v1alpha1"
+	v1_storage "k8s.io/api/storage/v1"
+)
+
+func TestGetCreateCASTemplate(t *testing.T) {
+	sc := &v1_storage.StorageClass{}
+	sc.Annotations = make(map[string]string)
+	tests := map[string]struct {
+		scCreateCASAnnotation string
+		scCASTypeAnnotation   string
+		envJivaCAST           string
+		envCStorCAST          string
+		expectedCAST          string
+	}{
+		"CAST annotation is present": {
+			"cast-create-from-annotation",
+			"",
+			"",
+			"",
+			"cast-create-from-annotation",
+		},
+		"CAST annotation is absent/empty and cas type is cstor": {
+			"",
+			"cstor",
+			"",
+			"cast-cstor-create-from-env",
+			"cast-cstor-create-from-env",
+		},
+		"CAST annotation is absent/empty and cas type is jiva": {
+			"",
+			"jiva",
+			"cast-jiva-create-from-env",
+			"",
+			"cast-jiva-create-from-env",
+		},
+		"CAST annotation is absent/empty and cas type unknown": {
+			"",
+			"unknown",
+			"cast-jiva-create-from-env",
+			"cast-cstor-create-from-env",
+			"",
+		},
+	}
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			sc.Annotations[string(v1alpha1.CASTemplateKeyForSnapshotCreate)] = test.scCreateCASAnnotation
+			sc.Annotations[string(v1alpha1.CASTypeKey)] = test.scCASTypeAnnotation
+			os.Setenv(string(menv.CASTemplateToCreateCStorSnapshotENVK), test.envCStorCAST)
+			os.Setenv(string(menv.CASTemplateToCreateJivaSnapshotENVK), test.envJivaCAST)
```

Comment:
As we setting an env better to unset it at the end of test ...using defer()...
we can improve it upcoming PR's.